src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -1294,6 +1294,13 @@ public virtual string Labels
         {
             get
             {
+                var labels = _configurationManagerStatic.GetAppSetting("NewRelic.Labels");
```

Comment:
A pattern we're moving towards (in the interest of performance) is caching values like this in private class members after they're calculated the first time to prevent repeatedly querying ConfigurationManager and Environment Variables. 

In this case, could you introduce a private member named `_labels`, to avoid un-necessarily querying ConfigurationManager after the first time this Property getter is called?